vad_transcriber/audioTranscript_gui.py

```diff
@@ -267,29 +298,115 @@ def transcriptionStart_on_click(self):
         # Execute
         self.threadpool.start(worker)
 
+    @pyqtSlot()
+    def CheckConvert_SR_on_click(self):  
+        logging.debug('check and convert to 16000')
+        
+        print("self wave file",self.fileName)
+        print("\n")
+        
+       # dst = "E:\\STT_DS_Mozilla_Work\\deepspeech\\DeepSpeech-examples-r0.7\\vad_transcriber\\converted_16k\\r1.wav"
+        dst = './Converted_16K_SR_audio/a1.wav'
+        samplerate, data = wavfile.read(self.fileName)
+        
+        print("Current wave file sample rate is :",samplerate)
+        print("\n")
+        #print("  data:" , data)
+        
+        inrate=samplerate
+        outrate=16000 
+        inchannels=1 
+        outchannels=1
+            
+        if not os.path.exists(self.fileName):
+            print('Source not found!')
+            return False
+    
+        if not os.path.exists(os.path.dirname(dst)):
+            os.makedirs(os.path.dirname(dst))
+        
+        try:
+            s_read = wave.open(self.fileName, 'r')
+            s_write = wave.open(dst, 'w')
+            
+            
+        except:
+            print('Failed to open files!')
+            return False
+    
+        n_frames = s_read.getnframes()
+        #print("n_frames",n_frames)
+        #print("\n")
+        data = s_read.readframes(n_frames)
+        #print("data : ",data)
+        
+        #samplerate, data = wavfile.read('/content/output.wav')
+        
+        samplerate, data = wavfile.read(self.fileName)
+        
+       
+        
+        try:
+            converted = audioop.ratecv(data, 2, inchannels, inrate, outrate, None)
```

Comment:
I think you should rather use inspiration from existing code for performing this step: https://github.com/mozilla/DeepSpeech/blob/0b007c9fb6df56fa277e4f9278e9340af59db2b4/native_client/python/client.py#L22-L31